\chapter{Introduction}\label{sec:introduction}
% Intro to drug development
Currently, pharmacological drug development focuses on well-established biochemistry based approaches to find and optimize new drugs. However, the challenges these methods face are manifold. High cost related to drug failure rates during various clinical trials and commercialization bottleneck the industry. Another important aspect is the occurrence of adverse drug reactions subjecting patients to hospitalization, possibly ending up fatal. Therefore, the pharmaceutical industry is not only facing high financial risks but also humanitarian issues that strain the trust-based relationship between the industry, physicians and patients.\cite{Katara2013}\\
% Computational drug development now
Academia demonstrated computational tools to be employable to many health industry challenges, such as costs of drug target validation, drug safety, and commercialization.\cite{Myers2001}\cite{Nelson2008} Albeit, chemo- and bioinformatics are novel and complex disciplines recently fostered by technological advancements in high-throughput methods and big-data analysis. Thus, the health industry does not yet benefit from promises like computer-aided identification of drugs and drug targets on a large scale.\\
% Cell Painting
New high-throughput methods and automated microscopy gave rise to the development of high-content imaging, which is frequently used to record small compound perturbations inflicted on biological systems. High-content-imaging applies up to six fluorescent dyes allowing to portray up to five different compartments per cell simultaneously. This method also referred to as \acf{cp} captures compound perturbed biological systems and automatically resolves cellular characteristics.  Computational models can interpret these in the context of a morphological fingerprint (or \ac{cp} feature vectors).\cite{Bray2017,Simm2017}\\
% Cell Profiler
The raw images from high-content imaging are processed, mostly by the software CellProfiler\cite{Carpenter2006} which extracts up to \num{1800} numerical features per image (e.g. nucleus shape, \ac{er} texture, etc.). Features from \ac{cp} assays can be interpreted as a morphological fingerprint, unique for each compound.\cite{Simm2017} By now, many different \ac{cp} assays have been conducted. A widely used \ac{cp} data set was generated by Bray \textit{et al}.\cite{Bray2016} The images were recorded with sixfold fluorescence staining for imaging five crucial cellular organelles (further information see \fref{sec:cpassay}).\cite{Bray2017} The concept of \ac{cp} is visualized in \fref{fig:cpprocess}.\\
\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{figures/cp_process.png}
	\caption[Visualization of a \ac{cp} Assay]{Visualization of a \ac{cp} assay. First cellular images are generated by fluorescence microscopy, then cellular objects and compartments are recognized by CellProfiler from which a large data table is generated containing the morphological fingerprint for each compound. Figure from Rohban \textit{et al.} and Carpenter \textit{et al.}, modified.\cite{Carpenter2006,Rohban2017}}
	\label{fig:cpprocess}
\end{figure}\noindent
% Gustafsdottir
Recently, Gustafsdottir \textit{et al.}\cite{Gustafsdottir2013} used \ac{cp} data to link morphological states to \acp{moa} via gene expression data. Hierarchical clustering was used to find clusters of compounds, addressing the same set of genes and, therefore, the same biochemical pathway. The results obtained from this \ac{cp} based approach mirrored the findings in the literature, which showed that \ac{cp} data is directly correlated to cellular pathways responding to compound perturbations.\\
% Nassiri and McCall
Nassiri and McCall\cite{Nassiri2018} used different \ac{cp} assay data\cite{Wawer2014a} and compound perturbed gene expression data in the context of machine learning methods. They used a LASSO model to predict cell morphological features against similar gene expression profiles. In-depth analysis of the results revealed strong model predictiveness among compounds that steer gene expression in the same direction, suggesting common \acp{moa}. In addition to linking \ac{cp} data to \acp{moa} they revealed direct relations between compounds' \acp{moa} based on machine learning model performance.\\
% Rohban et al
Furthermore, Rohban \textit{et al}.\cite{Rohban2017} transduced U-2 OS cells with lentiviral particles carrying cDNA constructs for gene overexpression.\cite{Wiemann2016,Yang2011} In their approach, they conducted a \ac{cp} assay to annotate the overexpressed genes with morphological fingerprints (numerical \ac{cp} features). After calculating the Pearson correlation between each morphological fingerprint, they used hierarchical clustering resulting in \num{25} clusters for \num{110} overexpressed genes. The clusters generated from the \ac{cp} data agglomerated genes that correspond to similar or identical pathways showing that genes can be connected using relatively inexpensive \ac{cp} assays. Furthermore, they predicted an unknown relationship between the Hippo- and NF-$\kappa$B-pathway.\\
% lapins and Spjuth
Lapins and Spjuth\cite{Lapins2019} annotated compounds of the \ac{cp} data from Bray \textit{et al.}\cite{Bray2017} and from the CMap\cite{Subramanian2017} (gene expression profiles) with their \ac{moa} or target protein. The information about the compound-wise \ac{moa} and targets was obtained from the Drug Repurposing Hub or the Touchstone database. In total, they annotated \num{1484} compounds present in \ac{cp} and CMap data with \num{234} \acp{moa} or targets. As the third set of descriptors, structural fingerprints were generated. For several targets and \acp{moa} a trained \ac{rfc} could present significant discriminatory power (\acs{auc}>\num{0.7}). Furthermore, it was found that the three different descriptors were complementary to a certain degree, each excelling at different \acp{moa} or targets. Lapins and Spjuth not only showed that \ac{cp} data could be used to predict compound's \ac{moa} but also that a combination with other identifiers is likely to enhance their applicability domain.\\
% Simm et al
Simm \textit{et al}.\cite{Simm2017} studied a \ac{cp} assay specifically designed for \ac{gcr} nuclear translocation. After treating H4 brain neuroglioma cells with \num{524371} compounds, hydrocortisone was added to stimulate \ac{gcr} translocation. Next, the treated cells were stained, imaged and processed analogously to the work of Gustafsdottir \textit{et al.}\cite{Gustafsdottir2013} The \num{524371} compounds were not only annotated with morphological information, but also with target activity information from \num{600} biochemical assays. Noticeably, most compounds were covered in few assays only, amounting to a fill rate of \SI{1.6}{\percent}. From this sparse activity matrix, they built a \ac{ml} model with \ac{cp} data as side information to predict all labels within the activity matrix. They evaluated the discriminatory power of their model and \num{34} bioassays (out of \num{600}) showed high predictivity (\acs{auc}>\num{0.9}). One of the assays was part of an ongoing discovery project. Within this assay, the highest-ranking \num{342} compounds, by matrix factorization, were experimentally tested. \num{141} ($41.2\%$) of these resulted in submicromolar hits which means a \num{60}-fold enrichment over the initial \ac{hts}. Another assay with an \ac{auc} greater \num{0.9} was part of an ongoing drug discovery project and could achieve a \num{250}-fold hit enrichment over the initial \ac{hts} in an analogous way. Their work presented \ac{cp} data as highly informative descriptors that might be repurposed for the prediction of sparse activity matrices. Additionally, they demonstrated their potential in ongoing drug discovery projects.\cite{Simm2018}\\
% working in data science and with CP
Data science in drug development and pharmacology has great potential. However, some caveats require carefulness when working with \ac{cp} data to access drug safety. The first one is imbalanced data. An assay testing compounds on a potential target will always feature fewer actives than inactive. Chawla \textit{et al.}\cite{Chawla2002} proposed a technique that mitigates this effect called \ac{smote}. This technique allows generating synthetic data that fit into the distribution of the real data points. Another obstacle is the high dimensionality of \ac{cp} data. Only a comparably small number of features contains most of the information necessary to predict a given target. Thus, statistical tools are applied for sophisticated feature selection in \ac{cp} studies. A \ac{cp} data set containing too many features is bound to overfit the data and give overoptimistic predictions on the test set without generalizing particularly well. The project above of Rohban \textit{et al.}\cite{Rohban2017} tackled this problem by \ac{pca}. Thereby reducing the number of features from \num{2769} to \num{158} which comprise most of the variance.\\
% about this work
In this explorative \ac{ml} project, targets obtained from bioassays published on \acl{p} are predicted with regard to descriptors from the \ac{cp} data set of Bray \textit{et al.}\cite{Bray2017}. The \acl{p} assays are selected based on their relation to targets presumably contributing to cytotoxicity.\cite{Mervin2016} The results are compared to the small molecules' structural fingerprints, and the performance metrics are analyzed extensively. From this analysis, conclusions can be drawn, whether and when to use \ac{cp} data. Insights gained from this project might guide future decision making when it comes to the prediction of biological endpoints.